src/compiler/commandLineParser.ts

```diff
@@ -771,6 +777,17 @@ namespace ts {
         }
     }
 
+    export function convertCompileOnSaveOptionFromJson(jsonOption: any, basePath: string, errors: Diagnostic[]): boolean {
+        if (!hasProperty(jsonOption, "compileOnSave")) {
+            return false;
+        }
+        const result = convertJsonOption({ name: "compileOnSave", type: "boolean" }, jsonOption["compileOnSave"], basePath, errors);

```

Comment:
extract `{ name: "compileOnSave", type: "boolean" }` into a variable so it can be used here and in `commandLineParser.ts`